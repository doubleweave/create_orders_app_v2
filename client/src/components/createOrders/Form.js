import React, { useState, useEffect } from 'react'
import TotalAmount from './subForm/TotalAmount'
import Consumer from './subForm/Consumer'
import Billing from './subForm/Billing'
import Shipping from './subForm/Shipping'
import Items from './subForm/Items'
import Merchant from './subForm/Merchant'

import { Box } from '@mui/material'
import { styled } from '@mui/material/styles'
import LinearProgress, {
  linearProgressClasses,
} from '@mui/material/LinearProgress'

import { useLocation, useHistory } from 'react-router-dom'
import { Router, Route, Switch } from 'react-router'

const pageStyle = {
  width: {
    xs: '100%',
    sm: '380px',
    xl: '380px',
  },
  border: {
    xs: '1px yellow solid',
    sm: '1px red solid',
    xl: '1px green solid',
  },
  height: '100vh',
  margin: '0 auto',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  alignItems: 'space-between',
  flexGrow: 1,
  flexShrink: 1,
  // maxHeight: '100vh',
  // // overflow: 'hidden',
  // overflowY: 'scroll',
}

const formTopLabel = {
  width: '100%',
  height: '50px',
  textAlign: 'center',
  lineHeight: '50px',
}

const BorderLinearProgress = styled(LinearProgress)(({ theme }) => ({
  height: 10,
  borderRadius: 5,
  [`&.${linearProgressClasses.colorPrimary}`]: {
    backgroundColor:
      theme.palette.grey[theme.palette.mode === 'light' ? 200 : 800],
  },
  [`& .${linearProgressClasses.bar}`]: {
    borderRadius: 5,
    backgroundColor: theme.palette.mode === 'light' ? '#1a90ff' : '#308fe8',
  },
}))

const routes = {
  '/totalAmount': () => <TotalAmount />,
  '/consumer': () => <Consumer />,
  '/billing': () => <Billing />,
  '/shipping': () => <Shipping />,
  '/items': () => <Items />,
}

function Form() {
  // const [page, setPage] = useState(0)
  const location = useLocation()
  const history = useHistory()

  console.log('location in Form: ', location)
  console.log('history in Form: ', history)

  const FormTitle = [
    '/totalAmount',
    '/consumer',
    '/billing',
    '/shipping',
    '/items',
    '/merchant',
  ]

  const [progressWidth, setProgressWidth] = useState(0)

  let path = location.pathname

  useEffect(() => {
    console.log('path in Form page: ', path)

    if (path === '/') {
      history.replace('/totalAmount')
    }

    let page = FormTitle.findIndex((item) => item === path)
    if (page !== -1) {
      setProgressWidth((page + 1) * (100 / 6))
    }
  }, [path])

  return (
    <Box className="form" sx={pageStyle}>
      <div>
        <Box
          color="secondary"
          sx={{
            width: '92%',
            height: '15px',
            left: 2,
            mt: 2,
            ml: 2,
          }}
        >
          <BorderLinearProgress
            data-testid="linearProgress"
            variant="determinate"
            value={progressWidth}
          />
        </Box>
      </div>
      <div>
        <Switch>
          <Route path="/totalAmount" component={TotalAmount} />
          <Route path="/consumer" component={Consumer} />
          <Route path="/billing" component={Billing} />
          <Route path="/shipping" component={Shipping} />
          <Route path="/items" component={Items} />
          <Route path="/merchant" component={Merchant} />
        </Switch>
      </div>
    </Box>
  )
}

export default Form
