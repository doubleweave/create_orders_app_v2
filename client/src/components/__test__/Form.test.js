import { render, screen, cleanup } from '@testing-library/react'
import ShallowRenderer from 'react-test-renderer/shallow'

import Form from '../createOrders/Form'

function mockFunction() {
  return {
    useHistory: () => ({
      push: jest.fn(),
    }),
    useLocation: () => ({
      pathname: '/totalAmount',
    }),
  }
}

jest.mock('react-router-dom', () => mockFunction())

describe('Testing Form', () => {
  it('Renders correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<Form />)
    const result = renderer.getRenderOutput()

    expect(result).toMatchSnapshot()
  })
})
