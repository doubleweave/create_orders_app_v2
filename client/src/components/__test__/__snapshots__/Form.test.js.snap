// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Form Renders correctly 1`] = `
<ForwardRef(Box)
  className="form"
  sx={
    Object {
      "alignItems": "space-between",
      "border": Object {
        "sm": "1px red solid",
        "xl": "1px green solid",
        "xs": "1px yellow solid",
      },
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "height": "100vh",
      "justifyContent": "space-between",
      "margin": "0 auto",
      "width": Object {
        "sm": "380px",
        "xl": "380px",
        "xs": "100%",
      },
    }
  }
>
  <div>
    <ForwardRef(Box)
      color="secondary"
      sx={
        Object {
          "height": "15px",
          "left": 2,
          "ml": 2,
          "mt": 2,
          "width": "92%",
        }
      }
    >
      <Styled(ForwardRef(LinearProgress))
        data-testid="linearProgress"
        value={0}
        variant="determinate"
      />
    </ForwardRef(Box)>
  </div>
  <div>
    <Switch>
      <Route
        component={[Function]}
        path="/totalAmount"
      />
      <Route
        component={[Function]}
        path="/consumer"
      />
      <Route
        component={[Function]}
        path="/billing"
      />
      <Route
        component={[Function]}
        path="/shipping"
      />
      <Route
        component={[Function]}
        path="/items"
      />
      <Route
        component={[Function]}
        path="/merchant"
      />
    </Switch>
  </div>
</ForwardRef(Box)>
`;
